src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,56 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+
 public class WorkWithFile {
+    private static final String COMMA = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
 
     public void getStatistic(String fromFileName, String toFileName) {
+        writeReportToFile(toFileName, readDataFromFile(fromFileName));
+    }
+
+    private String[] readDataFromFile(String fromFileName) {
+        try {
+            File fromFile = new File(fromFileName);
+            return Files.readAllLines(fromFile.toPath()).toArray(new String[0]);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from the file: " + fromFileName, e);
+        }
+    }
+
+    private void writeReportToFile(String toFileName, String[] strings) {
+        try {
+            File toFile = new File(toFileName);
+            toFile.createNewFile();
+            String[] resultString = resultString(strings);
```

Comment:
you shouldn't call this method here, you should do this in getStatistic() and pass in current method already formed report